pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotTableRestletResourceTest.java

```diff
@@ -960,6 +960,32 @@ public void testTableWithSameNameAsLogicalTableIsNotAllowed()
         aThrows.getMessage());
   }
 
+  @Test
+  public void testGetNonExistentTableConfig() {
+    // Attempt to get a non-existent table config
+    String tableName = "nonExistentTable";
+    String url = DEFAULT_INSTANCE.getControllerRequestURLBuilder().forTableGet(tableName);
+    String msg = expectThrows(IOException.class, () -> sendGetRequest(url)).getMessage();
+    assertTrue(
+        msg.contains("Got error status code: 404 (Not Found) with reason: \"Table nonExistentTable does not exist\""),
+        msg);
+
```

Comment:
nit: Could we add a variant of `sendGetRequest` that also returns a status code and makes an assertion on that instead?

Reply:
See https://github.com/apache/pinot/blob/e18845e4b93bfaa0548052846369b5b323dc5514/pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java#L1045 for instance.